tests/recordRequests.js

```diff
@@ -3,53 +3,71 @@ const fs = require(`node:fs`);
 const path = require(`node:path`);
 const v8 = require(`node:v8`);
 
-const nock = require(`nock`);
+/**
+ * @type {Map<string, {body: string, status:number, headers: Record<string,string>}> | undefined}
+ */
+let mocks;
 
 const getNockFile = () =>
   path.join(
     __dirname,
     `nock`,
     `${process.env.NOCK_FILE_NAME}-${process.env.RUN_CLI_ID}.dat`,
   );
-const ACCEPTED_HEADERS = new Set([`Content-Type`, `Content-Length`]);
-function filterHeaders(headers) {
-  if (!Array.isArray(headers)) return headers;
 
-  const filtered = [];
-  for (let t = 0; t < headers.length; t += 2)
-    if (ACCEPTED_HEADERS.has(headers[t].toLowerCase()))
-      filtered.push(headers[t], headers[t + 1]);
+if (process.env.NOCK_ENV === `record`) {
+  const realFetch = globalThis.fetch;
+  globalThis.fetch = async (input, init) => {
+    const response = await realFetch(input, init);
+    const data = await response.arrayBuffer();
 
-  return filtered;
-}
+    const minimalHeaders = new Headers();
+    for (const headerName of [`content-type`, `content-length`]) {
+      const headerValue = response.headers.get(headerName);
+      if (headerValue != null) {
+        minimalHeaders.set(headerName, headerValue);
+      }
+    }
 
-switch (process.env.NOCK_ENV || ``) {
-  case `record`:
-    nock.recorder.rec({
-      // eslint-disable-next-line @typescript-eslint/naming-convention
-      dont_print: true,
-      // eslint-disable-next-line @typescript-eslint/naming-convention
-      output_objects: true,
+    mocks ??= new Map();
+    mocks.set(input.toString(), {
+      body: Buffer.from(data).toString(`base64`),
```

Comment:
We definitely need a comment here, otherwise someone inevitably going to wonder why do we bother with base64.